src/app/static/src/tests/components/projects/shareProject.test.ts

```diff
@@ -39,236 +39,259 @@ describe("ShareProject", () => {
+            props: {
                 project: {id: 1, name: "p1"}
             },
-            store,
+            global: {
+                plugins: [store]
+            }
         });
 
         const link = wrapper.find("button");
-        link.trigger("click");
-        const input = wrapper.find(Modal).find("input");
-        input.setValue("Test.user@Example.com");
-        input.trigger("blur");
-        await Vue.nextTick();
-        const modal = wrapper.find(Modal);
+        await link.trigger("click");
+        const input = wrapper.findComponent(Modal).find("input");
+        await input.setValue("Test.user@Example.com");
+        await input.trigger("blur");
+        await nextTick();
+        const modal = wrapper.findComponent(Modal);
         expect(modal.find("input").classes()).toContain("is-invalid");
 
         const text = modal.find(".text-danger")
         expect(text.classes()).not.toContain("d-none");
-        expectTranslated(text, "Projects cannot be shared with the user's own account",
+        await expectTranslated(text, "Projects cannot be shared with the user's own account",
             "Les projets ne peuvent pas être partagés avec le propre compte de l'utilisateur",
             "Os projetos não podem ser partilhados com a conta do próprio utilizador", store as any)
-        expect(modal.find("button").attributes("disabled")).toBe("disabled");
+        expect(modal.find("button").attributes("disabled")).toBe("");
         expect(modal.find(".help-text").isVisible()).toBe(true);
     });
 
     it("if a valid email is entered twice, it is invalid, appropriate validation feedback is shown, and button is disabled", async () => {
         const store = createStore(jest.fn().mockResolvedValue(true))
-        const wrapper = mount(ShareProject, {
-            propsData: {
+        const wrapper = mountWithTranslate(ShareProject, store, {
+            props: {
                 project: {id: 1, name: "p1"}
             },
-            store,
+            global: {
+                plugins: [store]
+            }
         });
 
         const link = wrapper.find("button");
-        link.trigger("click");
-        const modal = wrapper.find(Modal);
+        await link.trigger("click");
+        const modal = wrapper.findComponent(Modal);
         expect(modal.findAll("input").length).toBe(1);
         const input = modal.find("input");
-        input.setValue("goodemail");
-        input.trigger("blur");
-        await Vue.nextTick();
+        await input.setValue("goodemail");
+        await input.trigger("blur");
+        await nextTick();
 
         expect(modal.findAll("input").length).toBe(2);
-        const input2 = wrapper.find(Modal).findAll("input").at(1);
+        const input2 = wrapper.findComponent(Modal).findAll("input")[1];
         input2.setValue("goodemail");
-        input2.trigger("blur");
+        await input2.trigger("blur");
 
-        setTimeout(() => {
+        setTimeout(async () => {
```

Comment:
Yh I have just removed all `setTimeout`s in this file. I prefer async await anyway and cannot really see much advantage of `setTimeout`s in any of our tests!

Note: two tests actually failed when I removed the setTimeouts, they are "if email is valid, validation feedback is not shown and button enabled" and "if email is valid but cased differently, validation feedback is not shown and button enabled" tests. The reason was that they were looking for `.text-danger` and seeing if it had a `d-none` class (bootstrap display none) but this failed because we had it wrapped in a `v-if` so `.text-danger` was not being rendered at all, ive fixed these tests now!